03_Modules/Configuration/src/module/api.ts

```diff
@@ -88,26 +101,43 @@ export class ConfigurationModuleApi
     super(ConfigurationComponent, server, logger);
   }
 
+
   /**
    * Message Endpoint Methods
    */
 
   @AsMessageEndpoint(
     CallAction.SetNetworkProfile,
     SetNetworkProfileRequestSchema,
+    { websocketServerConfigId: { type: 'string' } }
   )
-  setNetworkProfile(
+  async setNetworkProfile(
     identifier: string,
     tenantId: string,
     request: SetNetworkProfileRequest,
     callbackUrl?: string,
+    extraQueries?: Record<string, any>
   ): Promise<IMessageConfirmation> {
+    const correlationId = uuidv4();
+    if (extraQueries) {
+      const websocketServerConfigId = extraQueries[SetNetworkProfileExtraQuerystrings.websocketServerConfigId];
+      await SetNetworkProfile.build({
+        stationId: identifier,
+        correlationId,
+        configurationSlot: request.configurationSlot,
+        websocketServerConfigId,
+        apn: JSON.stringify(request.connectionData.apn),
+        vpn: JSON.stringify(request.connectionData.vpn),
+        ...request.connectionData
+      }).save();
```

Comment:
Just a question, in this way the create CrudEvent will not be emitted but it is also not necessary to emit SetNetworkProfile and related db entries, right? 🤔

Reply:
Yes!